android/src/main/java/com/instabug/flutter/modules/ApmApi.java

```diff
@@ -0,0 +1,237 @@
+package com.instabug.flutter.modules;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.instabug.apm.APM;
+import com.instabug.apm.model.ExecutionTrace;
+import com.instabug.apm.networking.APMNetworkLogger;
+import com.instabug.flutter.generated.ApmPigeon;
+import com.instabug.flutter.util.ArgsRegistry;
+import com.instabug.flutter.util.Reflection;
+
+import org.json.JSONObject;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.flutter.plugin.common.BinaryMessenger;
+
+public class ApmApi implements ApmPigeon.ApmHostApi {
+    private final String TAG = ApmApi.class.getName();
+    private final HashMap<String, ExecutionTrace> traces = new HashMap<>();
+
+    public static void init(BinaryMessenger messenger) {
+        final ApmApi api = new ApmApi();
+        ApmPigeon.ApmHostApi.setup(messenger, api);
+    }
+
+    @Override
+    public void setEnabled(@NonNull Boolean isEnabled) {
+        new Handler(Looper.getMainLooper()).post(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    APM.setEnabled(isEnabled);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
I would prefer if we did that in a separate PR _(as you mentioned above)_ that improves error handling, to avoid any side effects, considering the size of this PR.

Reply:
I've added a backlog item for it.